### PR TITLE
feat: add vector retrieval and update policy/template

### PR DESCRIPTION
**Pull request by pedrocassalpacheco**
This modifies the module configuration to reflect the proper class name, modifies the AstraDBGraphVectorComponent to support different search algorithms, and fixes the JSON parsing for the index policy property. It also adds a template for Graph RAG.

### CODE REVIEW

File: src/backend/base/langflow/components/vectorstores/astradb_graph.py [hunk 1]
```diff
@@ -248,10 +256,17 @@ def _add_documents_to_vector_store(self, vector_store) -> None:
             logger.debug("No documents to add to the Vector Store.")
 
     def _map_search_type(self) -> str:
+        if self.search_type == "Similarity":
+            return "similarity"
         if self.search_type == "Similarity with score threshold":
             return "similarity_score_threshold"
         if self.search_type == "MMR (Max Marginal Relevance)":
             return "mmr"
+            return "similarity_score_threshold"
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/components/vectorstores/astradb_graph.py:265]: There are two returns here.

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
Hey @pedrocassalpacheco 

Thanks for the improvements. 
Could you review the PR title to conform with conventional commits (https://www.conventionalcommits.org/en/v1.0.0/#summary) ?

File: src/backend/base/langflow/components/vectorstores/astradb_graph.py (hunk 1 above)
Comment (pedrocassalpacheco) [reply src/backend/base/langflow/components/vectorstores/astradb_graph.py:265]: Oi Gabriel - certainly, There is also a regression issue that pre-dates me. Looks like the class name on the __init.py__ differs from the astradbvectorgraphcomponent. Should I change that as well so the unit tests pass?

**Comment by pedrocassalpacheco** [comment]
> Hey @pedrocassalpacheco
> 
> Thanks for the improvements. Could you review the PR title to conform with conventional commits (https://www.conventionalcommits.org/en/v1.0.0/#summary) ?

@ogabrielluiz - can you review this again? I think I addressed your concerns ...

**Review by erichare — APPROVED** [review]

**Comment by pedrocassalpacheco** [comment]
@ogabrielluiz - I removed the template from the PR as requested

**Pull request merged**